torch/csrc/jit/register_c10_ops.cpp

```diff
@@ -0,0 +1,66 @@
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <torch/csrc/jit/operator.h>
+
+namespace torch {
+namespace jit {
+namespace {
+
+// TODO This currently only handles tensors with requires_grad==False correctly.
+//      It should also handle autograd.
+Operator createOperatorFromC10(const c10::OperatorHandle& op) {
+  return Operator(op.schema(), [op](Stack& stack) {
+      const auto input_size = op.schema().arguments().size();
+      const auto output_size = op.schema().returns().size();
+
+      // unwrap tensor inputs from variable
```

Comment:
This seems like an incredibly strange way to do unwrapping. Wouldn't it make more sense for this to happen at the dispatch stage? From first principles, I'd expect the dispatcher to take Variables.

Reply:
c10 dispatcher also has to work when calling caffe2 kernels from the caffe2 frontend, in which case it can't get a variable. This is why the current approach unwraps before entering the dispatcher.

This point is probably fixed once Variable goes away.